pwiz_tools/Skyline/TestRunnerLib/RunTests.cs

```diff
@@ -349,8 +347,18 @@ public bool Run(TestInfo test, int pass, int testNumber, string dmpDir, bool hea
                 LocalizationHelper.CurrentCulture = LocalizationHelper.CurrentUICulture = Language;
                 LocalizationHelper.InitThread();
 
+                // Unit tests normally don't do disk access, so don't mess around with tempdir creation for those
+                var exerciseTMP = test.TestClassType?.BaseType?.Name != "AbstractUnitTest" &&
+                                  test.TestClassType?.BaseType?.Name != "AbstractUnitTestEx";
+                if (exerciseTMP)
```

Comment:
Seems cleaner to just use the test in this if clause and then test for tmpTestDir != null after that. Currently you have 2 if clauses below, one which reuses this value and one which tests for tmpTestDir != null.

Reply:
Yeah, that's tighter.